kubetest2-tf/data/vpc/main.tf

```diff
@@ -1,38 +1,27 @@
-data "ibm_is_vpc" "vpc" {
-  count = var.vpc_name == "" ? 0 : 1
-  name  = var.vpc_name
+data "ibm_resource_group" "default_group" {
+  name = var.vpc_resource_group
 }
 
-data "ibm_is_subnet" "subnet" {
-  count = var.vpc_subnet_name == "" ? 0 : 1
-  name  = var.vpc_subnet_name
+data "ibm_is_image" "node_image" {
+  name = var.node_image
 }
 
-data "ibm_resource_group" "default_group" {
-  name = var.vpc_resource_group
+data "ibm_is_ssh_key" "ssh_key" {
+  name = var.vpc_ssh_key
 }
 
 module "vpc" {
-  # Create new vpc and subnet only if vpc_name is not set
-  count          = var.vpc_name == "" ? 1 : 0
   source         = "./vpc-instance"
+  vpc_name       = var.vpc_name
```

Comment:
I am not entirely convinced with the design here, IMO we are just delegating the decision making to another TF file, maybe we should move bits related to VPC creation out here (and perform new or existing logic here) and just pass relevant IDs to the nested module.

@Rajalakshmi-Girish VPC module here wont be affecting ppc related operations right?

Reply:
@Prajyot-Parab Yes I don't think the changes here will effect our ppc related operations.
They use a different ppc module.